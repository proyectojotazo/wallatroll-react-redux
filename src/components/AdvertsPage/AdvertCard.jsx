import { Card, Button, ListGroup } from "react-bootstrap";

const AdvertCard = ({ ad, history }) => {
  const img = ad.photo
    ? `${process.env.REACT_APP_API_BASE_URL}${ad.photo}`
    : "https://via.placeholder.com/150x100?text=No Image";

  const handleRedirect = () => {
    // Redireccionar a la pagina del anuncio
    history.push(`/advert/${ad.id}`);
  };

  return (
    <Card className="my-2">
      <Card.Img
        variant="top"
        src={img}
        style={{ height: "230px", objectFit: "cover" }}
      />
      <Card.Header>
        <Card.Title className="text-center">{ad.name}</Card.Title>
      </Card.Header>
      <Card.Body>
        <Card.Text>Price: {ad.price}â‚¬</Card.Text>
        <Card.Text>{ad.sale ? "Sell" : "Buy"}</Card.Text>
        <Card.Header className="text-center">Tags</Card.Header>
        <ListGroup variant="flush">
          {ad.tags.map((tag, i) => (
            <ListGroup.Item key={i}>{tag}</ListGroup.Item>
          ))}
        </ListGroup>
        <Button
          className="w-100 mt-3"
          variant="primary"
          onClick={handleRedirect}
        >
          Details
        </Button>
      </Card.Body>
    </Card>
  );
};

export default AdvertCard;
