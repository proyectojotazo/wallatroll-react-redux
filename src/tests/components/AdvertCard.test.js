import { render, fireEvent, screen } from "@testing-library/react";

import renderer from "react-test-renderer";
import AdvertCard from "./../../components/AdvertsPage/AdvertCard";

describe("<AdvertCard />", () => {
  const advert = {
    id: "1",
    createdAt: "12/12/2020",
    name: "Piano",
    photo: "/photo.jpg",
    sale: true,
    price: 120,
    tags: ["tag"],
  };
  const history = { push: jest.fn("param") };

  test("should match snapshot", () => {
    const tree = renderer
      .create(<AdvertCard ad={advert} history={history} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should call history.push", () => {
    render(<AdvertCard ad={advert} history={history} />);
    const redirectButton = screen.getByRole("button");
    fireEvent.click(redirectButton);
    expect(history.push).toHaveBeenCalled();
  });
});
