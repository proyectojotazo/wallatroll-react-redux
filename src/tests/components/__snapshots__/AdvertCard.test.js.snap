// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvertCard /> should match snapshot 1`] = `
<div
  className="my-2 card"
>
  <img
    className="card-img-top"
    src="http://localhost:3001/photo.jpg"
    style={
      Object {
        "height": "230px",
        "objectFit": "cover",
      }
    }
  />
  <div
    className="card-header"
  >
    <div
      className="text-center card-title h5"
    >
      Piano
    </div>
  </div>
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      Price: 
      120
      â‚¬
    </p>
    <p
      className="card-text"
    >
      Sell
    </p>
    <div
      className="text-center card-header"
    >
      Tags
    </div>
    <div
      className="list-group list-group-flush"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        tag
      </div>
    </div>
    <button
      className="w-100 mt-3 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Details
    </button>
  </div>
</div>
`;
